misk/src/main/kotlin/misk/client/HttpClientsConfig.kt

```diff
@@ -125,45 +85,30 @@ data class HttpClientConfig(
   val readTimeout: Duration? = null,
   val pingInterval: Duration? = null,
   val callTimeout: Duration? = null,
-  val maxRequests: Int = 128,
```

Comment:
Given that this data structure is now used for both defaults and overrides - all fields need to be null. Otherwise, merging an override on top of default would always override all non-nullable fields.